### PR TITLE
conversion to saml

### PR DESCRIPTION
**Pull request by skarunakar**
@kazukisan  

We are trying to modify this gem and publish, as we are  not owners of gem unable to publish this. Can you please review, merge and publish the gem
Can also make me as owner?

### CODE REVIEW

**Comment by GiantEnemyCrab** [comment]
I haven't looked into the code base for a long time but how does this compare to another pull request, as usertoken seems to be the focus of both ?

https://github.com/kalinchuk/quickbase/pull/8

File: lib/classes/http.rb [hunk 1]
```diff
@@ -11,9 +11,8 @@ def initialize(config)
       http_proxy = config[:http_proxy] || ENV['http_proxy']
       setup_proxy(http_proxy) if http_proxy
 
-      qb_params[:ticket] = auth_ticket config
-      qb_params[:apptoken] = config[:apptoken]
       qb_params[:dbid] = config[:dbid]
+      qb_params[:usertoken] = config[:usertoken]
     end
```
Comment (skarunakar) [inline lib/classes/http.rb:16]: @kalinchuk 
Usertoken can be used with any API that doesn’t post to a db/main URL, and also with API_GrantedDBs.
This method initialize is called for API that does not post to db/main URL ; for eg API like getSchema, API_DoQuery etc.
For those API query must be like mentioned here 
https://help.quickbase.com/api-guide/create_user_tokens.html

**Comment by skarunakar** [comment]
@kazukisan 

In pull request #8  using just usertoken  without intializing in file connection.rb will pass empty tags  without any values .
Also this change in QB was announced in oct 2016 and by end of this month gem has to be published.
Please help us in meeting the requirements before deadline.

File: lib/classes/connection.rb [hunk 2]
```diff
@@ -30,7 +30,8 @@ def instantiate
         apptoken: apptoken,
         dbid: dbid,
         org: org,
-        http_proxy: http_proxy
+        http_proxy: http_proxy,
+        usertoken: usertoken
```
Comment (skarunakar) [inline lib/classes/connection.rb:34]: Without this empty tags <usertoken></usertoken> will be passed in query

**Comment by skarunakar** [comment]
@kazukisan  Thanks for merging the changes, as you are one of the owners , can you version this and publish in rubygems.org

**Comment by GiantEnemyCrab** [comment]
@skarunakar, I am comfortable with the idea of making you one of the owners and you can publish it. Let me know how that sounds.

**Comment by skarunakar** [comment]
@kazukisan  That works, Please make me as one of the owners.
username - skarunakar
Thank you.

**Comment by GiantEnemyCrab** [comment]
I realized that I do not have the privilege to do so. Sorry about it.
But @kalinchuk should be able to do so.

**Comment by kalinchuk** [comment]
@skarunakar should be able to push now.

**Comment by skarunakar** [comment]
@kalinchuk @kazukisan   I am ok if I dont get privilege for pushing to this repo. I asked for making me as owner of this gem.
When i run command gem owner quickbase - I see this 
Owners for gem: quickbase
- a**@gmail.com
- k**@gmail.com
- j***@gmail.com

If their is an entry - s***r@intuit.com that would be great. 
If this is not possible, we are anyway  working with legal team at intuit to get this repo  into github.com/intuit so that we can publish gem on own on

**Comment by GiantEnemyCrab** [comment]
That's great to know! I hope the process goes well on that!
(On the side note, I went ahead and edited the comment to mask personal email addresses, but once your email addition is done, that can be edited, too)

**Comment by kalinchuk** [comment]
@skarunakar I added you as owner of gem.

**Pull request merged**